plugins/modules/nmcli.py

```diff
@@ -2130,8 +2137,12 @@ def _compare_conn_params(self, conn_info, options):
 
             if isinstance(current_value, list) and isinstance(value, list):
                 # compare values between two lists
-                if sorted(current_value) != sorted(value):
-                    changed = True
+                if key in ('ipv4.addresses', 'ipv6.addresses'):
+                    # The order of IP addresses matters because the first one
+                    # is the default source address for outbound connections.
+                    changed = current_value != value
+                else:
+                    changed = sorted(current_value) != sorted(value)
```

Comment:
This part seems to fail the unit tests.

Not sure why...I would suggest to move this to another PR and make the origin changes for this PR commited?